src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -93,8 +93,15 @@ static HRESULT _duplicateHandle(const HANDLE in, HANDLE& out) noexcept
 //   from the registered handler event function.
 HRESULT CTerminalHandoff::EstablishPtyHandoff(HANDLE in, HANDLE out, HANDLE signal, HANDLE ref, HANDLE server, HANDLE client) noexcept
 {
+    // Stash a local copy of _pfnHandoff before we stop listening.
+    auto localPfnHandoff = _pfnHandoff;
+
+    // Because we are REGCLS_SINGLEUSE... we need to `CoRevokeClassObject` after we handle this ONE call.
+    // COM does not automatically clean that up for us. We must do it.
+    s_StopListening();
+
     // Report an error if no one registered a handoff function before calling this.
-    RETURN_HR_IF_NULL(E_NOT_VALID_STATE, _pfnHandoff);
+    RETURN_HR_IF_NULL(E_NOT_VALID_STATE, localPfnHandoff);

```

Comment:
Should this line be _above_ the `s_StopListening` call? Or should we always stop listening, even if this is null. 


I suppose, if this is null, then we do want to stop being the server for inbound connections first, otherwise we'll leave ourself polluted. 

okay this makes sense